### PR TITLE
Add {Get,Update}VMMetadata calls to the fc-control api.

### PR DESCRIPTION
**Pull request by IRCody**
This enables callers of the fc-control api  to patch
metadata instead of requiring a user to submit all the
metadata everytime.

Signed-off-by: Cody Roseborough <cdr@amazon.com>

*Issue #, if available:* #248

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: runtime/service_integ_test.go [hunk 1]
```diff
@@ -1003,3 +1003,58 @@ func TestDriveMountFails_Isolated(t *testing.T) {
...
+
+	rootfsPath := defaultVMRootfsPath
+	_, err = fcClient.CreateVM(ctx, &proto.CreateVMRequest{
+		VMID: "1",
+		MachineCfg: &proto.FirecrackerMachineConfiguration{
+			MemSizeMib: 512,
+		},
+		RootDrive: &proto.FirecrackerRootDrive{
+			HostPath: rootfsPath,
+		},
```
Comment (sipsma) [inline runtime/service_integ_test.go]: nit: no need to specify this, the runtime's default is the same value

File: runtime/service_integ_test.go [hunk 2]
```diff
@@ -1003,3 +1003,58 @@ func TestDriveMountFails_Isolated(t *testing.T) {
 		assert.Error(t, err, "unexpectedly succeeded in creating a VM with a drive mount under banned path")
 	}
 }
+
+func TestUpdateVMMetadata_Isolated(t *testing.T) {
```
Comment (sipsma) [inline runtime/service_integ_test.go]: I think there would be some value in adding a container to this test that gets the VM metadata too so we can verify that we are able to spawn containers that have access to it too. There's an existing test case that uses `wget` in the alpine container image which could probably be adapted to hit the mmds endpoint instead: https://github.com/firecracker-microvm/firecracker-containerd/blob/38742eeb4fb5d549b31727a57b8c79e88b204b59/runtime/cni_integ_test.go#L112

Comment (IRCody) [reply runtime/service_integ_test.go]: That's a good idea, I'll try to add it.

File: runtime/service_integ_test.go (hunk 1 above)
Comment (IRCody) [reply runtime/service_integ_test.go]: I tried removing this but it fails to create the vm if I do, saying it tries to stat the hostpath and it's "". I changed it to not use rootfsPath and just use the defaultVMRootFSPath directly though.

File: runtime/service_integ_test.go (hunk 2 above)
Comment (IRCody) [reply runtime/service_integ_test.go]: Added this based on the example you linked.

**Pull request merged**